<!DOCTYPE HTML>
<html lang="en" class="sidebar-visible no-js light">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>Genome Annotation Guide</title>
        <meta name="robots" content="noindex" />


        <!-- Custom HTML head -->
        
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff" />

        <link rel="icon" href="favicon.svg">
        <link rel="shortcut icon" href="favicon.png">
        <link rel="stylesheet" href="css/variables.css">
        <link rel="stylesheet" href="css/general.css">
        <link rel="stylesheet" href="css/chrome.css">
        <link rel="stylesheet" href="css/print.css" media="print">

        <!-- Fonts -->
        <link rel="stylesheet" href="FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="fonts/fonts.css">

        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" href="highlight.css">
        <link rel="stylesheet" href="tomorrow-night.css">
        <link rel="stylesheet" href="ayu-highlight.css">

        <!-- Custom theme stylesheets -->

    </head>
    <body>
        <!-- Provide site root to javascript -->
        <script>
            var path_to_root = "";
            var default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "navy" : "light";
        </script>

        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script>
            try {
                var theme = localStorage.getItem('mdbook-theme');
                var sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script>
            var theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            var html = document.querySelector('html');
            html.classList.remove('no-js')
            html.classList.remove('light')
            html.classList.add(theme);
            html.classList.add('js');
        </script>

        <!-- Hide / unhide sidebar before it is displayed -->
        <script>
            var html = document.querySelector('html');
            var sidebar = 'hidden';
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            }
            html.classList.remove('sidebar-visible');
            html.classList.add("sidebar-" + sidebar);
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <div class="sidebar-scrollbox">
                <ol class="chapter"><li class="chapter-item expanded "><a href="about.html"><strong aria-hidden="true">1.</strong> About This Book</a></li><li class="chapter-item expanded "><a href="quick_guide.html"><strong aria-hidden="true">2.</strong> Quick Guide</a></li><li class="chapter-item expanded "><a href="quality_control.html"><strong aria-hidden="true">3.</strong> Quality Control</a></li><li class="chapter-item expanded "><a href="advanced_annotation.html"><strong aria-hidden="true">4.</strong> Advanced Annotation</a></li><li class="chapter-item expanded "><a href="changelog.html"><strong aria-hidden="true">5.</strong> Changelog</a></li></ol>
            </div>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle"></div>
        </nav>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky bordered">
                    <div class="left-buttons">
                        <button id="sidebar-toggle" class="icon-button" type="button" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </button>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="light">Light</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search. (Shortkey: s)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="S" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">Genome Annotation Guide</h1>

                    <div class="right-buttons">
                        <a href="print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>

                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>

                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script>
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <h1 id="about-this-book"><a class="header" href="#about-this-book">About This Book</a></h1>
<p>As genome assemblies become more routine in everyday lab tasks, it is important to understand how to quickly do a good annotation for the assembly, as predicted gene models can affect many downstream analyses (RNA-Seq, Phylogenomics, etc).</p>
<p>Genome annotation is a rapidly changing field, with new pipelines, tools, methods, and techniques released with somewhat regular frequently. It is not a solved problem at this stage.</p>
<p>Further, we explore how to examine the quality of genome annotation gene models.</p>
<h2 id="updates"><a class="header" href="#updates">Updates</a></h2>
<p>As this field is always changing, this book is a work-in-progress. It should be considered pre-release alpha at this stage.</p>
<h2 id="genome-annotation"><a class="header" href="#genome-annotation">Genome Annotation</a></h2>
<p>Annotation can mean multiple things. Here Genome Annotation is the annotation to find gene models for the sequence of the assembly. Separately, annotation of the gene models (rather than the assembly) is what we use to give gene names, GO categories, and other metadata. Please refer to the diagrams below.</p>
<pre class="mermaid">graph TD
    A[Genome Assembly] --&gt;|Annotation| B(Putative Gene Models)
</pre>
<pre class="mermaid">graph TD
    B[Gene Models] --&gt;|Annotation| C(Gene Models)
    C --&gt; D[Putative Gene Names]
    C --&gt; E[Gene Ontology Terms]
    C --&gt; F[Motifs]
</pre>
<h2 id="non-gene-annotations"><a class="header" href="#non-gene-annotations">Non-Gene Annotations</a></h2>
<p>New tools and databases have made it easier to annotate other important genomic features, such as repeats, transposable elements, long non-coding RNA (lncRNA), and others. These will be touched on as well.</p>
<div style="break-before: page; page-break-before: always;"></div><h1 id="quick-guide"><a class="header" href="#quick-guide">Quick Guide</a></h1>
<h2 id="what-data-do-you-have"><a class="header" href="#what-data-do-you-have">What data do you have?</a></h2>
<pre class="mermaid">%%{init: { 
    'logLevel': 'debug', 
    'theme': 'forest', 
    'gitGraph': {'showBranches': true, 'showCommitLabel':true,'mainBranchName': 'Assembly'},
    'themeVariables': {
        'commitLabelFontSize': '16px'
    }
} }%%
gitGraph
   commit id: &quot;Assembly&quot;
   commit id: &quot;Predicted Proteins&quot;
   commit id: &quot;GALBA&quot;
   commit id: &quot;Have RNA-Seq?&quot;
   branch RNA-Seq
   checkout RNA-Seq
   commit id: &quot;BRAKER2&quot;
   checkout Assembly
   commit id:&quot;Gene Models&quot; type:HIGHLIGHT
   checkout RNA-Seq
   merge Assembly tag: &quot;Combine&quot;
   commit type:REVERSE id: &quot;TSBERA&quot;
   commit id: &quot;Combined Gene Models&quot; type:HIGHLIGHT
</pre>
<p>The best tools at this time, from my experience, and proven by QC checks, are <a href="https://github.com/Gaius-Augustus/GALBA">GALBA</a> and <a href="https://github.com/Gaius-Augustus/BRAKER">BRAKER2</a> (soon BRAKER3).</p>
<h1 id="table-of-contents"><a class="header" href="#table-of-contents">Table of Contents</a></h1>
<ul>
<li><a href="quick_guide.html#galba">GALBA</a>
<ul>
<li><a href="quick_guide.html#checklist">Checklist</a></li>
<li><a href="quick_guide.html#download-and-install-galba">Download and Install GALBA</a>
<ul>
<li><a href="quick_guide.html#singularity">Singularity</a></li>
</ul>
</li>
<li><a href="quick_guide.html#download-uniprot-swissprot">Download UniProt SwissProt</a></li>
<li><a href="quick_guide.html#download-any-other-highquality-proteomes">Download any other HighQuality Proteome(s)</a></li>
<li><a href="quick_guide.html#run-galba">Run GALBA</a></li>
</ul>
</li>
<li><a href="quick_guide.html#braker2">BRAKER2</a></li>
<li><a href="quick_guide.html#quality-control">Quality Control</a>
<ul>
<li><a href="quick_guide.html#busco">BUSCO</a>
<ul>
<li><a href="quick_guide.html#install-busco">Install BUSCO</a></li>
<li><a href="quick_guide.html#run-busco-in-proteome-mode">Run BUSCO in Proteome Mode</a></li>
</ul>
</li>
<li><a href="quick_guide.html#genevalidator">GeneValidator</a>
<ul>
<li><a href="quick_guide.html#installation">Installation</a></li>
<li><a href="quick_guide.html#running-genevalidator">Running GeneValidator</a></li>
<li><a href="quick_guide.html#interpretation">Interpretation</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="quick_guide.html#gene-model-annotation">Gene Model Annotation</a>
<ul>
<li><a href="quick_guide.html#eggnog">EggNOG</a>
<ul>
<li><a href="quick_guide.html#installation-1">Installation</a></li>
<li><a href="quick_guide.html#running">Running</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="quick_guide.html#notes">Notes</a>
<ul>
<li><a href="quick_guide.html#why-the-version-files">Why the .version files?</a></li>
<li><a href="quick_guide.html#scripts">Scripts</a></li>
</ul>
</li>
<li><a href="quick_guide.html#final">Final</a></li>
</ul>
<h2 id="galba"><a class="header" href="#galba">GALBA</a></h2>
<p>GALBA does not take RNA-Seq nor transcripts as input, but only other gene models. You should use UniProt Swiss-Prot and, if available, a high-quality proteome from a related species (such as <em>Arabidopsis thaliana</em> for eudicots, for example). </p>
<h3 id="checklist"><a class="header" href="#checklist">Checklist</a></h3>
<p>The following are assumed</p>
<ul>
<li>Eukaryotic Assembly</li>
<li>Assembly is soft-masked</li>
<li>Contig/scaffold names are simple (no spaces, unusual characters, etc)</li>
</ul>
<h3 id="download-and-install-galba"><a class="header" href="#download-and-install-galba">Download and Install GALBA</a></h3>
<h4 id="singularity"><a class="header" href="#singularity">Singularity</a></h4>
<pre><code class="language-shell">singularity build galba.sif docker://katharinahoff/galba-notebook:devel
</code></pre>
<h3 id="download-uniprot-swissprot"><a class="header" href="#download-uniprot-swissprot">Download UniProt SwissProt</a></h3>
<pre><code class="language-shell">wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz
md5sum uniprot_sprot.fasta.gz &gt; uniprot.md5sum
date &gt; uniprot.download_date
gunzip uniprot_sprot.fasta.gz
</code></pre>
<h3 id="download-any-other-highquality-proteomes"><a class="header" href="#download-any-other-highquality-proteomes">Download any other HighQuality Proteome(s)</a></h3>
<p>Once done, add them all to a single file</p>
<pre><code class="language-shell">cat uniprot_sprot.fasta &gt; proteins.fasta
cat OTHER_FILE.fasta &gt;&gt; proteins.fasta
</code></pre>
<h3 id="run-galba"><a class="header" href="#run-galba">Run GALBA</a></h3>
<pre><code class="language-shell">singularity run galba.sif galba.pl --version &gt; galba.version
singularity exec --bind .:/data galba.sif galba.pl \
        --species=&quot;Myspecies Name&quot; \
        --genome=&quot;/data/assembly.fa&quot; \
        --prot_seq=/data/proteins.fa \
        --AUGUSTUS_ab_initio \
        --threads 8 \
        --softmasking \
        --workingdir=/data/galba \
        --gff3 \
        --crf
</code></pre>
<p>To additionally create an annotation without protein evidence, add </p>
<pre><code class="language-shell"></code></pre>
<p>to the command.</p>
<h2 id="braker2"><a class="header" href="#braker2">BRAKER2</a></h2>
<p>coming soon</p>
<h2 id="quality-control"><a class="header" href="#quality-control">Quality Control</a></h2>
<h3 id="busco"><a class="header" href="#busco">BUSCO</a></h3>
<p>BUSCO is a good check for single-copy orthologs, but is not a good metric for other gene models. However, it is a fast, necessary, first-step in QC. It should be close to the results for the genomic BUSCO (-m genome). BUSCO overpredicts when using metaeuk, and thus you should use --augustus when possible for genomic BUSCO runs, along with the (most appropriate) --augustus_species.</p>
<h4 id="install-busco"><a class="header" href="#install-busco">Install BUSCO</a></h4>
<h5 id="singularity-1"><a class="header" href="#singularity-1">Singularity</a></h5>
<pre><code class="language-shell">singularity build busco_5.4.4_cv1.sif docker://ezlabgva/busco:v5.4.4_cv1
</code></pre>
<h4 id="run-busco-in-proteome-mode"><a class="header" href="#run-busco-in-proteome-mode">Run BUSCO in Proteome Mode</a></h4>
<pre><code class="language-shell">singularity exec busco_5.4.4_cv1.sif busco --version &gt; busco.version

singularity exec --bind .:/data ../busco_5.4.4_cv1.sif busco \
	-m protein -i /data/galba/augustus.hints.aa \
	-o /data/buscos/busco_galba_augustus_hints \
	--auto-lineage-euk \
	--cpu 16
</code></pre>
<p>Replace --auto-lineage-euk with -l aves or the name of the BUSCO database you wish to compare to.</p>
<h3 id="genevalidator"><a class="header" href="#genevalidator">GeneValidator</a></h3>
<p>GeneValidator compares multiple metrics to a known-good database. For this it is best to use your newly downloaded UniProt Swiss-Prot database, and to not include other proteomes you may have added.</p>
<h4 id="installation"><a class="header" href="#installation">Installation</a></h4>
<p>Please see the (GitHub page)[https://github.com/wurmlab/genevalidator].</p>
<h4 id="running-genevalidator"><a class="header" href="#running-genevalidator">Running GeneValidator</a></h4>
<p>Running time is usually a few days for 30k proteins.</p>
<pre><code class="language-shell">genevalidator --version &gt; genevalidator.version
genevalidator -d uniprot_sprot.fasta --num_threads 32 -m 8 \
        -o genevalidator/galba_uniprot_augustus_hints \
        galba/augustus.hints.aa
</code></pre>
<p>You can additionally run it against the high-quality proteomes you have downloaded:</p>
<pre><code class="language-shell">genevalidator -d other_genome_file.fasta --num_threads 32 -m 8 \
        -o genevalidator/galba_SPECIESNAME_augustus_hints \
        galba/augustus.hints.aa
</code></pre>
<h4 id="interpretation"><a class="header" href="#interpretation">Interpretation</a></h4>
<p>There are multiple files to interpret, but the main file will be <strong>augustus.hints_summary.csv</strong>.</p>
<p>The output will look similar to:</p>
<pre><code class="language-csv">num_predictions,31842
num_good_predictions,8327
num_bad_predictions,21164
num_predictions_with_insufficient_blast_hits,6259
first_quartile_of_scores,22.0
second_quartile_of_scores,67.0
third_quartile_of_scores,90.0
</code></pre>
<p>(Note: These numbers do not add up because they are obfuscated from an upcoming paper).</p>
<p>A poorer genome annotation could look something like this:</p>
<pre><code class="language-csv">num_predictions,18294
num_good_predictions,4218
num_bad_predictions,16731
num_predictions_with_insufficient_blast_hits,5925
first_quartile_of_scores,0.0
second_quartile_of_scores,45.0
third_quartile_of_scores,67.0
</code></pre>
<p>Where the number of good predictions is much lower, and the score quartiles are generally lower as well.</p>
<p>GeneValidator gives results such as this (See: <strong>augustus3_results.csv</strong>):</p>
<pre><code class="language-csv">AnalysisNumber,GVScore,Identifier,NumberOfHits,LengthCluster,LengthRank,GeneMerge,Duplication,MissingExtraSequences
1,67,g1.t1,234,1172 [559 - 1134],1% (too long),0.1,1.0,96% conserved; 18% extra; 3% missing.
2,0,g2.t1,4,Not enough evidence,Not enough evidence,Not enough evidence,Not enough evidence,Not enough evidence
3,64,g3.t1,28,223 [190 - 509],11% (too short),-0.1,1.0,91% conserved; 0% extra; 25% missing.
4,45,g4.t1,6,337 [407 - 464],0% (too short),0.0,1.0,Not enough evidence
5,45,g5.t1,163,379 [460 - 500],5% (too short),0.0,1.0,88% conserved; 2% extra; 24% missing.
</code></pre>
<p>Easier to view in the table below:</p>
<div class="table-wrapper"><table><thead><tr><th>GVScore</th><th>NumberOfHits</th><th>LengthCluster</th><th>LengthRank</th><th>GeneMerge</th><th>Duplication</th><th>MissingExtraSequences</th></tr></thead><tbody>
<tr><td>67</td><td>234</td><td>1172 [559 - 1134]</td><td>1% (too long)</td><td>0.1</td><td>1.0</td><td>96% conserved; 18% extra; 3% missing</td></tr>
<tr><td>0</td><td>4</td><td>Not enough evidence</td><td>Not enough evidence</td><td>Not enough evidence</td><td>Not enough evidence</td><td>Not enough evidence</td></tr>
<tr><td>64</td><td>28</td><td>223 [190 - 509]</td><td>11% (too short)</td><td>-0.1</td><td>1.0</td><td>91% conserved; 0% extra; 25% missing</td></tr>
<tr><td>45</td><td>6</td><td>337 [407 - 464]</td><td>0% (too short)</td><td>0.0</td><td>1.0</td><td>Not enough evidence</td></tr>
<tr><td>45</td><td>163</td><td>379 [460 - 500]</td><td>5% (too short)</td><td>0.0</td><td>1.0</td><td>88% conserved; 2% extra; 24% missing</td></tr>
</tbody></table>
</div>
<p>As you can see, GeneValidator looks for gene models fitting an expected length distribution, gene merges, duplication, as well as extra sequence and missing sequence. As Swiss-Prot is a manually annotated list of &quot;known-good&quot; protein models, it is a high-quality source to check against.</p>
<h1 id="gene-model-annotation"><a class="header" href="#gene-model-annotation">Gene Model Annotation</a></h1>
<h2 id="eggnog"><a class="header" href="#eggnog">EggNOG</a></h2>
<p>EggNOG is the fastest way to get from gene models to gene names and GO terms. </p>
<h3 id="installation-1"><a class="header" href="#installation-1">Installation</a></h3>
<p>...coming soon...</p>
<h3 id="running"><a class="header" href="#running">Running</a></h3>
<p>...coming soon...</p>
<h1 id="notes"><a class="header" href="#notes">Notes</a></h1>
<h2 id="why-the-version-files"><a class="header" href="#why-the-version-files">Why the .version files?</a></h2>
<p>This makes it easier to track which version of a tool you used for an analysis, to put into your manuscript methods (or supplemental), and increase the reproducibility of the research.</p>
<h2 id="scripts"><a class="header" href="#scripts">Scripts</a></h2>
<p>You should place all of your commands into various scripts (.sh or .bash files) so that the exact command used to run part of the methods is preserved, and this should be a part of your GitHub release with the paper.</p>
<h1 id="final"><a class="header" href="#final">Final</a></h1>
<p>With these QC scores (BUSCO and GeneValidator), you should be able to tell the quality of your annotation and should now, quickly, have a workable annotation. Depending on your use case, there are a multitude of ways to improve the gene models, including additional RNA-Seq, or adding proteins of interest from related organisms you know to be high-quality to your protein input into the file (as opposed to entire proteomes from other species). </p>
<p>For further gene model annotation, please see the appropriate chapter in this book.</p>
<p>If higher-quality annotations are required, please read the rest of this guide!</p>
<div style="break-before: page; page-break-before: always;"></div><h1 id="quality-control-1"><a class="header" href="#quality-control-1">Quality Control</a></h1>
<div style="break-before: page; page-break-before: always;"></div><h1 id="advanced-annotation"><a class="header" href="#advanced-annotation">Advanced Annotation</a></h1>
<div style="break-before: page; page-break-before: always;"></div><h1 id="changelog"><a class="header" href="#changelog">Changelog</a></h1>
<h2 id="19-dec-2022"><a class="header" href="#19-dec-2022">19 Dec 2022</a></h2>
<p>Initial Work</p>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->


                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">

            </nav>

        </div>




        <script>
            window.playground_copyable = true;
        </script>


        <script src="elasticlunr.min.js"></script>
        <script src="mark.min.js"></script>
        <script src="searcher.js"></script>

        <script src="clipboard.min.js"></script>
        <script src="highlight.js"></script>
        <script src="book.js"></script>

        <!-- Custom JS scripts -->
        <script src="mermaid.min.js"></script>
        <script src="mermaid-init.js"></script>

        <script>
        window.addEventListener('load', function() {
            window.setTimeout(window.print, 100);
        });
        </script>

    </body>
</html>
